### PR TITLE
feat(GridForm): rowspan fields 

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview
Adds `rowspan` prop to GridFormFields for layout customization

### PR Checklist

- [x] Related to designs: [Profiles V0.2 Figma](https://www.figma.com/file/PG0kVYDIuLk3hE566beZUV/Profiles-(Concept-Refinement)?node-id=998%3A11232)
- [x] Related to JIRA ticket: EGG-1268
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.5-alpha.0791ed.0<br/>@codecademy/gamut-labs@18.3.5-alpha.0791ed.0<br/>@codecademy/gamut@33.4.3-alpha.0791ed.0<br/>@codecademy/styleguide@40.3.3-alpha.0791ed.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60b69dfcdbcffd273f5057b3--gamut-preview.netlify.app>https://60b69dfcdbcffd273f5057b3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60b69dfcdbcffd273f5057b3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.5-alpha.377fd6.0<br/>@codecademy/gamut-labs@18.3.5-alpha.377fd6.0<br/>@codecademy/gamut@33.4.3-alpha.377fd6.0<br/>@codecademy/styleguide@40.3.3-alpha.377fd6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60b6a0d4b013222d4f1c89a0--gamut-preview.netlify.app>https://60b6a0d4b013222d4f1c89a0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60b6a0d4b013222d4f1c89a0>Deploy Logs</a></p>

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx [hunk 1]
```diff
@@ -194,6 +194,74 @@ If an input is not meant to be usable, such as when a portion of a form is disab
   </Story>
 </Canvas>
 
+## GridForm-atting
+
+We can use the `size` and `rowspan` props (borrowed from [LayoutGrid](https://react-hook-form.com)) to customize the layouts of our GridForms.
+
+<Canvas>
+  <Story name="rowspanformatting">
```
Comment (katiezutter) [inline packages/styleguide/stories/Organisms/GridForm.stories.mdx:202]:
```suggestion
  <Story name="Rowspan Formatting">
```

The teeniest of nits! Just so it fits in nicely with the rest:

Screen Shot 2021-06-01 at 4 26 40 PM

**Review by katiezutter — APPROVED** [review]
Awesome ☀️

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.5-alpha.d47a1f.0<br/>@codecademy/gamut-labs@18.3.5-alpha.d47a1f.0<br/>@codecademy/gamut@33.4.3-alpha.d47a1f.0<br/>@codecademy/styleguide@40.3.3-alpha.d47a1f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60b6a96c801dfc29f498d21c--gamut-preview.netlify.app>https://60b6a96c801dfc29f498d21c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60b6a96c801dfc29f498d21c>Deploy Logs</a></p>

**Review by codecaaron — APPROVED** [review]
💯

File: packages/styleguide/stories/Organisms/GridForm.stories.mdx [hunk 2]
```diff
@@ -194,6 +194,74 @@ If an input is not meant to be usable, such as when a portion of a form is disab
   </Story>
 </Canvas>
 
+## GridForm-atting
```
Comment (sooinc) [inline packages/styleguide/stories/Organisms/GridForm.stories.mdx:197]: 😄✨

**Review by sooinc — APPROVED** [review]
LGTM!!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.6-alpha.3ff61f.0<br/>@codecademy/gamut-labs@18.3.6-alpha.3ff61f.0<br/>@codecademy/gamut@33.4.4-alpha.3ff61f.0<br/>@codecademy/styleguide@40.3.4-alpha.3ff61f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60b794eff223313a8d7a8c95--gamut-preview.netlify.app>https://60b794eff223313a8d7a8c95--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60b794eff223313a8d7a8c95>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60b7edc739575441ff0383cf--gamut-preview.netlify.app>https://60b7edc739575441ff0383cf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60b7edc739575441ff0383cf>Deploy Logs</a></p>

**Pull request merged**